### ISSUE TITLE
[FEA] Cast cudf.series to category dtype

### ISSUE THREAD

**Issue by VibhuJawa** [rapidsai/cudf#1664]
**Is your feature request related to a problem? Please describe.**

I would like to use `cudf.series.astype('category')` function to cast a column both (numerical and non numerical) as `dtype='category'` .

**Additional context**
I would like to have the functionality as provided in pandas (shown below): 

```python
import pandas as pd
df = pd.DataFrame({'c_1':['a','b','c','b'],'c_2':[0,1,2,0]})
df['cat_c1']=df['c_1'].astype('category')
```

**Current Casting Method**
The below casts it as a nvstrings object. 
`gdf['c_1']= gdf['c_1'].astype(pd.core.dtypes.dtypes.CategoricalDtype)`

**Comment by beckernick** [comment]
The suggested current casting method is not a work around to converting a Series to the category dtype.

More generally, we need a clean way to convert a StringColumn to a CategoricalColumn. This may be messy, as our current CategoricalColumn implementation is coupled to the data as a Buffer, which is currently not the underlying `data` for StringColumns.

**Comment by VibhuJawa** [comment]
Current work-around:


```python
import numpy as np
from cudf.dataframe import Series, CategoricalColumn


def categorize_num_sr(num_sr):
    codes, cats = num_sr.factorize()
    encoded_sr = Series(CategoricalColumn(data=codes.data, categories=cats,ordered=False))
    return encoded_sr

cudf_sr = cudf.Series([1,2,0,0,2,3])
encoded_sr = categorize_num_sr(cudf_sr)
encoded_sr.to_pandas()
```

```python
0    1
1    2
2    0
3    0
4    2
5    3
dtype: category
Categories (4, int64): [0, 1, 2, 3]
```

**Comment by shwina** [comment]
This should be fixed now that #2342 is merged. @VibhuJawa could you please confirm that you no longer have issues with `astype`?

```
In [4]: import cudf as pd 
   ...: df = pd.DataFrame({'c_1':['a','b','c','b'],'c_2':[0,1,2,0]}) 
   ...: df['cat_c1']=df['c_1'].astype('category')                                                                                                                                                               

In [5]: print(df['cat_c1'])                                                                                                                                                                                     
0    a
1    b
2    c
3    b
Name: cat_c1, dtype: category
```

**Comment by VibhuJawa** [comment]
Yup, Can confirm it works, Please feel free to close.
